app/scripts/modules/core/src/pipeline/config/templates/v2/configurePipelineTemplateModalV2.controller.ts

```diff
@@ -0,0 +1,251 @@
+    this.initialize();
+  }
+
+  public initialize(): void {
+    const { exclude, name, template } = this.pipelineTemplateConfig;
+
+    const excluded = exclude || [];
+    this.state.inheritTemplateNotifications = !excluded.includes('notifications');
+    this.state.inheritTemplateParameters = !excluded.includes('parameters');
+    this.state.inheritTemplateTriggers = !excluded.includes('triggers');
+
+    this.pipelineName = name;
+    this.source = template.reference;
+
+    this.loadTemplate()
+      .then(() => {
+        this.groupVariableMetadata();
+        this.initializeVariables();
+        if (!this.variables.length) {
+          this.state.noVariables = true;
+        }
+      })
+      .then(() => (this.state.loading = false))
+      .catch(() => {
+        Object.assign(this.state, { loading: false, error: false, planErrors: null, loadingError: true });
+      });
+  }
+
+  public cancel(): void {
+    this.$uibModalInstance.dismiss();
+  }
+
+  public formIsValid(): boolean {
+    return this.variables.every(v => v.errors.length === 0);
+  }
+
+  public submit(): IPromise<void> {
+    const config = this.buildConfig();
+
+    return PipelineTemplateReader.getPipelinePlan(config)
+      .then(plan => {
+        const { parameterConfig, notifications, triggers } = plan;
+        const { inheritTemplateParameters, inheritTemplateNotifications, inheritTemplateTriggers } = this.state;
+
+        const configWithInheritedValues = {
+          ...config,
+          ...(inheritTemplateParameters && parameterConfig ? { parameterConfig } : {}),
+          ...(inheritTemplateNotifications && notifications ? { notifications } : {}),
+          ...(inheritTemplateTriggers && triggers ? { triggers } : {}),
+        };
+
+        this.$uibModalInstance.close({ plan, config: configWithInheritedValues });
+      })
+      .catch((response: IHttpPromiseCallbackArg<IPipelineTemplatePlanResponse>) => {
+        Object.assign(this.state, { loading: false, error: true, planErrors: response.data && response.data.errors });
+      });
+  }
+
+  public dismissError(): void {
+    Object.assign(this.state, { error: false, planErrors: null, loadingError: false });
+  }
+
+  public buildConfig(): IPipelineTemplateConfigV2 {
+    const {
+      application: { name: appName },
+      state: { inheritTemplateNotifications, inheritTemplateParameters, inheritTemplateTriggers },
+      pipelineName,
+      pipelineTemplateConfig,
+      source,
+    } = this;
+
+    const excludeConfig = [
+      ...(inheritTemplateParameters ? [] : ['parameters']),
+      ...(inheritTemplateNotifications ? [] : ['notifications']),
+      ...(inheritTemplateTriggers ? [] : ['triggers']),
+    ];
+
+    return {
+      ...(pipelineTemplateConfig || {}),
+      name: pipelineName,
+      application: appName,
+      variables: this.transformVariablesForPipelinePlan(),
+      exclude: excludeConfig,
+      ...PipelineTemplateV2Service.getPipelineTemplateConfigV2(source),
```

Comment:
The config structure for V2 is different from V1 and relies on the V2 service to get the correct template structure